### ISSUE TITLE
Delegate methods are not firing with FB pod

### ISSUE THREAD

**Issue by ccynn** [patchthecode/JTAppleCalendar#1015]
IMPORTANT - BEFORE CREATING YOUR ISSUE PLEASE READ THE FOLLOWING:

1. PLEASE STATE THE VERSION NUMBER you are using. 7.1.6
2. Are you using the latest version of JTAppleCalendar? Latest version is currently 7.1.6 - Yes
3. PROVIDE ENOUGH INFORMATION SO I CAN RECREATE THE PROBLEM.

Delegate methods are not firing on touch. `cellForItem` fires, but methods such as `didSelectDate` or `shouldSelectDate` do not fire on touch and am admittedly lost on why it doesn't work because it was working before..

ibCalendarDatasource and ibCalendarDelegate of the calendarView are linked up through storyboard. I recreated another calendar with the same code and setup, and am still receiving the same unresponsive state.

Will happily offer any detail I can provide to help find a solve. Thank you.

**Comment by patchthecode** [comment]
Can you just send the sample bugged app?
drag the zip file in the chat box.

**Comment by ccynn** [comment]
Sure thing, see attached. Let me know if there are any questions! Ignore some of the code that is just used to adapt the calendar for my needs. Appreciate you looking into this.


[DebugApp.zip](https://github.com/patchthecode/JTAppleCalendar/files/2990877/DebugApp.zip)

**Comment by patchthecode** [comment]
it fires on mine correctly.
try doing an XCode clean and then build? :/

**Comment by patchthecode** [comment]
btw, the latest version is 7.1.7

**Comment by ccynn** [comment]
yeah...so i found that it does actually work and fires in the test project. i just have no idea why it doesn't work in my existing project even though it's the exact same setup. the calendar's not responding to any gesture..

**Comment by patchthecode** [comment]
The only thing i can say is that maybe you did something in your project which is causing this behavior?  I do not have your project so i really do not know.

the best away to find it your self is to recreate the calendar in your project on a new view controller to make sure it works there. Test it function by function until it works. Then maybe this might tell you where something went wrong.

**Comment by ccynn** [comment]
appreciate your message.

so i've spent yesterday and today trying to figure it out an answer in this black hole. i deleted and reinstalled pods, created new projects, scoured the web, the works..

with that said, i've actually gotten to the point where i was able to recreate yesterday's file but with the unresponsive touch. moreover, what i then dug into further was some weird conflict with the FBLoginSDK installed pod. when i uninstall it from the podfile, the calendar works, and when it's installed it doesn't.

i have no idea why, what, how, but do you think you can help me from this point to find a solve and any clarity on the technicalities here?

im so lost and really need the calendar to work..
[DebugApp.zip](https://github.com/patchthecode/JTAppleCalendar/files/2994759/DebugApp.zip)

**Comment by ccynn** [comment]
Yeah there's definitely something conflicting alongside that pod. Specifically FBSDK pod version 4.42.0. When I do another pod install of version 4.39.1, the delegate fires correctly again.

Still have no idea why and an explanation/solve would be nice to be up-to-date on the latest versions, but I've reverted to the older version temporarily.

**Comment by Angie93** [comment]
Hi! I have an issue similar to yours, but my delegate method isn't fired sometimes. 

I can click most days on the calendar, but on some days (like 5, 12 or 13 on giff) the delegate event is not fired. Have you got any idea why it happens?

![calendar](https://media.giphy.com/media/orVyUIoIcJV30CGH9s/giphy.gif)

**Comment by patchthecode** [comment]
@Angie93 can you provide a zip file of the problem like the other dev did above?

**Comment by patchthecode** [comment]
@ccynn you issue doesnt make any sense at all O_O 

I have no clue why simply adding a totally separate pod is causing no delegates to fire.
There isnt even code any code from the other pod running.
The pod was just simply added and is sitting there doing nothing. I am very confused.

In the sample app you gave me, is there any code of the facebook pod runnning?

**Comment by patchthecode** [comment]
@ccynn 

ok, after debugging some more --> You have the calendar embedded in a NavigationController which you pushed like this
```swift
navigationController?.pushViewController(confirmMeetupDateVC, animated: true)
```

If you go to the storyboard, and make the `ScheduleMeetupDateViewController` the **first** viewController instead of the NavigationViewController, then every thing works well again.

I am still lost as to why putting it in a navigation for your project causes this behavior when FaceBook pod is added. It doesnt make any logical sense. Still investigating.

**Comment by patchthecode** [comment]
@ccynn keep in mind that this is confusing because the calendar is simply a UICollectionView. There is no magic code, or weird stuff. It is simply a UICollectionView.

**Comment by ccynn** [comment]
@patchthecode thanks for digging into this. im literally as stumped as you are and have zero clue. i don't even know how one goes about debugging this as im not necessarily that experienced..

i do need navigation controller flow though, and as the temporary workaround I just reverted back to an older version of the FB pod. As for whether or not any code is running for FB, I mean, again, I really can't say, but if I had some distant distant distant theorized morsel, it could be FB tracking user activity(?).

All said, 'why', 'what', and 'how' have been my only recurring themes when I ran into this issue.

**Comment by shashi-iitr** [comment]
This issue is occurring with my prod app too. Delegate methods are not working. As @ccynn suggested I degraded FB pods version to 4.39.1 and now it's working (with latest JTAppleCalendar version 7.1.7). So weird problem.

**Comment by patchthecode** [comment]
I have put the `Help Wanted` flag attached to this.
This bug is a mystery to me.

**Comment by Angie93** [comment]
Hi, sorry for the late reply, I let a zip just with the usage of my calendar, I don't know what can be happening
[Calendar.zip](https://github.com/patchthecode/JTAppleCalendar/files/3115643/Calendar.zip)

**Comment by blender** [comment]
@Angie93 any chance you figured out the problem?

**Comment by pauloec** [comment]
Ah. The wonders of FB SDK... it can actually break things depending on how you named your selectors. I've seen this happen with other libraries and solution is always to change name of functions that are related to delegates.

**Comment by patchthecode** [comment]
@pauloec What do you suggest I do?
It would seem to me that FB is at fault here 🤔

**Comment by pauloec** [comment]
@patchthecode i don't advise you to do anything, these problems are too difficult to be tackled due to their complexity and also happens very rarely. Each dev has to test it and see if it will work or not. Unless you want to go deep into FBSDK and check all their KVO calls.

**Comment by bananabastard** [comment]
@pauloec check https://github.com/patchthecode/JTAppleCalendar/pull/1062
it will fix the issue

**Comment by patchthecode** [comment]
I think this is fixed with FB pod 4.39.1 >  ?

**Comment by petekeller2** [comment]
I had the same issue with FBSDKLoginKit version 5.1.1. Upgrading to FBSDKLoginKit version 5.4.0 resolves this issue.
